README.markdown

```diff
@@ -87,31 +87,33 @@ If you want to use a standardized set of run stages for Puppet, `include stdlib`
     path => '/etc/sudoers',
     line => '%sudonopw ALL=(ALL) NOPASSWD: ALL',
   }
-  ```
+  ~~~
+
+##### Parameters
+All parameters are optional, unless otherwise noted.
+
+* `after`: Specifies the line after which Puppet will add any new lines. (Existing lines are added in place.) Valid options: String. Default: Undefined.
+* `ensure`: Ensures whether the resource is present. Valid options: 'present', 'absent'. Default: 'present'.
+* `line`: **Required.** Sets the line to be added to the file located by the `path` parameter. Valid options: String. Default: Undefined.
+* `match`: Specifies a regular expression to run against existing lines in the file; if a match is found, it is replaced rather than adding a new line. Valid options: String containing a regex. Default: Undefined.
+* `multiple`: Determines if `match` can change multiple lines. If set to false, an exception will be raised if more than one line matches. Valid options: 'true', 'false'. Default: Undefined.
+* `name`: **Required.** Sets the name to use as the identity of the resource.
```

Comment:
The `name` parameter is not required. It's only necessary to provide `name` if you want the resource namevar to be different from the supplied `title` of the resource.